### PR TITLE
Fix #67: Add missing scaladocs

### LINKED ISSUE #67: Add scaladoc to FutureRelation methods
# Issue

Add missing scaladocs

# Problem Description

- the following classes are missing method documentation:
  - `FutureRelation`
  - `RequestResponseProtocol`
  - `Dactor`

# Supporting Information

### PR DESCRIPTION
**Pull request by srfc**
Fixes #67

## Proposed Changes

  - see #67

## Related

  - n/a

### CODE REVIEW

**Comment by srfc** [comment]
I noted down that some methods of `Dactor` were missing documentation but now it doesn't seem like that is so. Do you still remember what might still be missing documentation?

File: adbms/src/main/scala/de/up/hpi/informationsystems/adbms/definition/FutureRelation.scala [hunk 1]
```diff
@@ -12,12 +12,42 @@ import scala.util.Try
 
 trait FutureRelation extends Relation with Immutable with Awaitable[Try[Seq[Record]]] {
 
+  /**
+    * Creates objects of type `B` from this Relation using `mapping` and pipes the resulting objects to given Actor
+    * `receiver` on successful Future completion.
+    * @param mapping  function used to create messages sent via messaging from this Relation
+    * @param receiver ActorRef of actor to pipe resulting messages to
+    * @tparam B       type of messages to be created
+    */
```
Comment (CodeLionX) [inline adbms/src/main/scala/de/up/hpi/informationsystems/adbms/definition/FutureRelation.scala]: You may add an example for the usage pattern:
```scala
val msgs: Map[Int, Customer.CustomerRequest] = _
val customers: FutureRelation = Dactor.askDactor(system, classOf[Customer], msgs)
customers.pipeAsMessageTo(relation => <SelfCompanion>.UpdateCustomersMsg(relation), self)
```
or somethings similar.

File: adbms/src/main/scala/de/up/hpi/informationsystems/adbms/definition/FutureRelation.scala [hunk 2]
```diff
@@ -12,12 +12,42 @@ import scala.util.Try
...
+    * Returns this FutureRelation's contents wrapped in a Future.
+    * @return the Future[Relation] which will contain this FutureRelation records
+    */
   def future: Future[Relation]
 
+  /**
+    * Creates a new Relation by applying the function `f` to this Relation.
+    * @param  f is the function to be applied
+    * @return a new Relation with `f` applied
+    */
```
Comment (CodeLionX) [inline adbms/src/main/scala/de/up/hpi/informationsystems/adbms/definition/FutureRelation.scala]: returns a new `FutureRelation` ;)

File: adbms/src/main/scala/de/up/hpi/informationsystems/adbms/protocols/RequestResponseProtocol.scala [hunk 3]
```diff
@@ -4,14 +4,27 @@ import de.up.hpi.informationsystems.adbms.definition.Record
 
 object RequestResponseProtocol {
 
+  /** A Request to a Dactor for which you expect a Response.
+    *
+    * Each Dactor that can reply to queries subclasses these traits with their own message types so one can match on
+    * expected message types
+    */
```
Comment (CodeLionX) [inline adbms/src/main/scala/de/up/hpi/informationsystems/adbms/protocols/RequestResponseProtocol.scala]: I'm missing the link to the usage pattern with `Dactor.askDactor()`

Comment (CodeLionX) [reply adbms/src/main/scala/de/up/hpi/informationsystems/adbms/protocols/RequestResponseProtocol.scala]: Or you link it the other way around: Link from `Dactor.askDactor()` to the messaging protocol

**Pull request merged**